osu.Game.Rulesets.Mania/Difficulty/Skills/README.md

```diff
@@ -0,0 +1,280 @@
+# Mania Strain Calculation Primer
+
+`StrainValueOf` governs how note strain is evaluated, which in-turn affects how star rating is calculated.
+
+---
+
+## Theory
+
+Strain refers to the difficulty of the note.
+
+There are 3 strain values present.
+
+- **Global Strain** $GS$: the overall running strain, depends on all previous notes.
+- **Column Strain** $CS$: the per-column running strain, depends on all previous notes in the same column.
+  - $CS_k$: the $k^{th}$ column running strain. 
+- **Strain** $S$: a function of GS and CS. $S=S(GS, CS)$
+
+### Intuition
+
+Take for example a simple strain calculator:
+
+- A 2K Map denoted on the columns K0, K1, with 3 notes, marked by X
+- Each note increments both strains by 5
+- Each time step decays both strain by 1 till 0
+- Strain is the sum of Global and the note's column-th Column Strain
+
+| Time | GS | CS0 | CS1 | S    | K0 | K1 | Desc.                                   |
+|------|----|-----|-----|------|----|----|-----------------------------------------|
+| -1   | 1  | 0   | 0   |      |    |    | Initialize Strains                      |
+| 0    | 6  |     | 5   | 6+5  |    | X  | 1st Note on Column 1, add to GS and CS1 |
+| 1    | 5  |     | 4   |      |    |    | Apply Decay                             |
+| 2    | 10 | 5   | 3   | 10+5 | X  |    | 2nd Note on Column 0, add to GS and CS0 |
+| 3    | 9  | 4   | 2   |      |    |    | Apply Decay                             |
+| 4    | 14 | 3   | 7   | 14+7 |    | X  | 3rd Note on Column 1, add to GS and CS1 |
+| 5    | 13 | 2   | 6   |      |    |    | Apply Decay                             |
+
+- GS increments on every note regardless of column, while CS only increments for their respective column.
+- S depends on GS and CS, never itself.
+    - On the 2nd note, S=GS+CS0 ignoring CS1 because the note's column is 0
+    - Same goes for the 3rd note, S=GS+CS1
+- CS is a matrix, GS is a vector.
+
+Though the actual script is non-trivial, it has similar concepts from the above example:
+
+- Our decay is [exponential](#evaluating-strains), so high strain values are exponentially rarer
+- We consider [Hold Interactions](#hold-strain-bonus-triggers), notes in complex Hold scenarios are have boosted
+  strains.
+- Decay is zero when evaluating notes at the same time (e.g. chords)
+- Strain is still the sum of Global and Column Strain
+
+### Strains
+
+We'll use math notations to define our strain robustly:
+
+- $GS_i$: GS on the $i^{th}$ object
+- $CS_{i,k}$: CS on the $i^{th}$ object on column $k$.
+- $S_i$: S on the $i^{th}$ object.
+- $O_i$: $i^{th}$ note/object
+- $\Delta_i$: time elapsed between the $i^{th}$ and previous object
+- $\Delta_{i,k}$: time elapsed between the $i^{th}$ and previous object in the same column $k$
```

Comment:
This is confusing. One delta has one subscript, one has two, and where there are two, then one of the subscripts is an object index and the other is a column index. Just declare delta with two subscripts as $\Delta t_{i,j} = t_j - t_i$ or something and then when you need the column strain thing, just declare an extra index for the last note in the same column and explain what it is locally to it.